apps/keyboard/test/marionette/input_test.js

```diff
@@ -148,4 +148,59 @@ marionette('Input with Keyboard APP', function() {
 
     assert.equal(inputString, keyboardTestApp.textInput.getAttribute('value'));
   });
+
+  test('Double tapping space bar in textarea', function() {
+    // Switch to test app frame.
+    client.switchToFrame();
+    client.apps.switchToApp(KeyboardTestApp.ORIGIN);
+
+    // Focus on a textarea
+    keyboardTestApp.textInput3.click();
+
+    client.switchToFrame();
+    systemInputMgmt.switchToActiveKeyboardFrame();
+
+    var inputString = 'Aa';
+    keyboard.type(inputString);
+
+    // type 2 spaces
+    var space = keyboard.getKey(' ');
+    space.click();
+    space.click();
+
+    // Switch to test app frame.
+    client.switchToFrame();
+    client.apps.switchToApp(KeyboardTestApp.ORIGIN);
+
+    assert.equal('As. ', keyboardTestApp.textInput3.getAttribute('value'));
+  });
+
+  test('tap space bar and then wait for a while before tapping again',
+      function() {
+    // Switch to test app frame.
+    client.switchToFrame();
+    client.apps.switchToApp(KeyboardTestApp.ORIGIN);
+
+    // Focus on a textarea
+    keyboardTestApp.textInput3.click();
+
+    client.switchToFrame();
+    systemInputMgmt.switchToActiveKeyboardFrame();
+
+    var inputString = 'Aa';
+    keyboard.type(inputString);
+
+    // type one space and then tap again after a while
+    var space = keyboard.getKey(' ');
+    space.click();
+
+    client.helper.wait(1000);
```

Comment:
The time to wait here is too long. I think 600ms would be long enough since we currently set it to 450ms in `ActiveTargetsManager`.